Parser.h

```diff
@@ -1,32 +1 @@
-#pragma once
-#include <string>
-#include <stack>
-#include "Node.h"
-#include "Grammar.h"
-
-class Parser
-{
-private:
-	Grammar grammar;
-	std::string cuvant;
-	char state;
-	int i;
-	std::stack<std::string> working_stack;
-	std::stack<char> input_stack;
-public:
-	Parser(Grammar g, std::string word);
-
-	void expand();
-	void advance();
-	void momentary_insuccess();
-	void back();
-	void another_try();
-	void success();
-
-	void construct_tree(Node *& node, std::stack<std::string> & stack);
-
-	std::string doamne_ajuta();

```

Comment:
Is this a magic method that you call when you get an exception and it somehow makes life better? What does it do?